Instant Search: Add the "Clear filters" button to the tab loop.

a8ck3n opened the pull request:
Just a standard looking button at the moment but it works across all browsers now and doesn’t break tabbing.

<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Addresses part of #19035.

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

The "Clear filters" button is now a `button` instead of an `a`. This allows it to participate in the tab loop in Firefox and no longer breaks the loop in other browsers.

**Note:** Just a standard button now. The old styling wasn't that nice but we can tweak that based on feedback.

Before:
![ClearOld](https://user-images.githubusercontent.com/40267301/181228535-0d4b4e66-7220-4cfa-8764-62ddb34d734a.png)

After:
![ClearNew](https://user-images.githubusercontent.com/40267301/181228563-eee38497-a240-45c9-92da-976b3c5b892a.png)

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

No

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

1. Open the Instant Search overlay.
2. Click on a filter or two so the "Clear filters" button appears.
3. Try tabbing to it.
4. Confirm that both Return and Space activate the button and clear the filters.

a8ck3n committed:
Add “Clear filters” button to tab loop.

Just a standard looking button at the moment but it works across all browsers now and doesn’t break tabbing.

kangzj reviewed (commented):
It works as described. However do you think it's better to keep the styling the same as it was ? It had been a button before it was changed to a link in #14762 where @keoshi provided design review.

a8ck3n committed:
Style the “Clear filters” button like a link.

Should handle hover & focus states correctly now.

a8ck3n commented:
> It works as described. However do you think it's better to keep the styling the same as it was? It had been a button before it was changed to a link […] provided design review.

Thanks for flagging those previous changes. I've updated the styling to match so it's once again looking like a link. Re-tested in the big 3 browsers and it looks pretty good.

dognose24 commented on projects/packages/search/src/instant-search/components/search-filters.scss:
```diff
@@ -24,6 +24,13 @@
+
+	&:hover,
+	&:focus {
+		text-decoration: underline;
```
I would suggest leaving text-decoration: none is appropriate to keep styles the same between the `sort` buttons and the `Clear filters` button for `:hover` and `:focus`.

<img width="265" alt="截圖 2022-08-03 上午3 26 13" src="https://user-images.githubusercontent.com/6869813/182457995-55163989-c185-46ed-874b-50698c1eb968.png">
<img width="574" alt="截圖 2022-08-03 上午3 32 17" src="https://user-images.githubusercontent.com/6869813/182458344-4645bf26-a801-431b-9a07-a184657dd7e4.png">

a8ck3n replied:
Ah, nice catch. Thanks @dognose24. I'll get that fixed up.

a8ck3n committed:
Update styles for “Clear filters” button…

…to more closely match the styles on the sort option buttons. Improves consistency on the overlay.

a8ck3n commented:
Thanks for the review @dognose24. I've updated the styles there to remove the `text-decoration` and add the `cursor` attribute. The two button styles are a little more consistent now. Would you mind giving this another quick look?

dognose24 reviewed (approved):
The sort and filter clearing button styles look consistent when focusing and hovering on browsers Safari, Firefox, and Chrome!

sdixon194 committed:
Merge remote-tracking branch 'origin/trunk' into fix/search-clear-filters-a11y-2
